src/approaches/interactive_learning_approach.py

```diff
@@ -116,29 +108,30 @@ def _relearn_predicates_and_operators(self) -> None:
             positive_examples = []
             negative_examples = []
             # Positive examples
-            for i, trajectory in enumerate(self._ground_atom_dataset):
-                for j, ground_atom_set in enumerate(trajectory):
-                    state = self._dataset[i][0][j]
+            for (states, _, ground_atom_sets) in self._dataset:
+                assert len(states) == len(ground_atom_sets)
+                for (state, ground_atom_set) in zip(states, ground_atom_sets):
+                    if len(ground_atom_set) == 0:
+                        continue  # pragma: no cover
```

Comment:
Is there a reason we can't test this? E.g. if we give data that contains states but we have no predicates, then ground_atom_set should always be empty, and we can test that there are no positives?

Reply:
good point! how is this?

Reply:
if the test is working correctly, we should be able to remove this "pragma: no cover" right?

Reply:
right, i forgot to remove it